CONTRIBUTING.md

```diff
@@ -0,0 +1,190 @@
+
+**NOTE:** Run `make help` for more information on all potential `make` targets
+
+
+## Getting Started
+You’ll need a Kubernetes cluster to run against. You can use 
+- [KIND](https://sigs.k8s.io/kind) 
+- [MINIKUBE](https://minikube.sigs.k8s.io)  
+- [CRC](https://console.redhat.com/openshift/create/local) 
+
+to get a local cluster for testing, or run against a remote cluster.
+**Note:** Your controller will automatically use the current context in your kubeconfig file (i.e. whatever cluster `kubectl cluster-info` shows).
+
+### Building
+
+```sh
+make container-build
+```
+
+
+### Modifying the API definitions
+If you are editing the API definitions, generate the manifests such as CRs or CRDs using:
+
+```sh
+make manifests
+```
+
+
+### Running on the cluster
+1. Install Instances of Custom Resources:
+
+```sh
+kubectl apply -f config/samples/
+```
+
+2. Build and push your image to the location specified by `IMG`:
+
+```sh
+make docker-build docker-push IMG=<some-registry>/kogito-serverless-operator:tag
+```
+
+3. Deploy the controller to the cluster with the image specified by `IMG`:
+
+```sh
+make deploy IMG=<some-registry>/kogito-serverless-operator:tag
+```
+
+### Uninstall CRDs
+To delete the CRDs from the cluster:
+
+```sh
+make uninstall
+```
+
+### Undeploy controller
+UnDeploy the controller to the cluster:
+
+```sh
+make undeploy
+```
+
+Minikube or CRC
+
+## Test the Greeting workflow on Minikube
+
+A good starting point to check that everything is working well, it is the [Greeting workflow](https://github.com/kiegroup/kogito-examples/blob/stable/README.md#serverless-workflow-getting-started).
+
+Follow these steps to create a container that you can than deploy as a Service on Kubernetes or KNative.
+
+1. Start Minikube
+```sh 
+minikube start --cpus 4 --memory 4096 --addons registry --insecure-registry "10.0.0.0/24"
+```
+2. Create a namespace for the building phase
+
+```sh
+kubectl create namespace kogito-builder
+```
+
+3. Create a secret
+```sh
+kubectl create secret docker-registry regcred --docker-server=<registry_url> --docker-username=<registry_username> --docker-password=<registry_password> --docker-email=<registry_email> -n kogito-builder
+```
+
+4. Build and push your image to the location specified by `IMG`:
+
+```sh
+make container-build container-push IMG=<some-registry>/kogito-serverless-operator:tag
+```
+
+5. Deploy the controller to the cluster with the image specified by `IMG`:
+
+```sh
+make deploy IMG=<some-registry>/kogito-serverless-operator:tag
+```
+
+6. Create a dedicated Namespace for the test:
+
+```sh
+kubectl create namespace greeting-workflow
+```
+
+7. Install Instances of Custom Resources:
+
+```sh
+kubectl apply -f config/samples/sw.kogito.kie.org__v08_kogitoserverlessworkflow.yaml -n greeting-workflow
+```
+
+8. Default configuration
+
+A configmap called kogito-serverless-operator-builder-config will be created under the kogito-serverless-operator-system
```

Comment:
from where this namespace is coming from?

Reply:
IS created by the operator, but we add the configmap under kogito-builder namespace

Reply:
ok